test/csv/csv-filter/test_csv_filter.rb

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+# frozen_string_literal: false
+
+require_relative '../helper'
+
+require 'csv'
+
+class TestFilter < Test::Unit::TestCase
+
+  # Some rows data (useful as default).
+  Rows = [
+    %w[aaa bbb ccc],
+    %w[ddd eee fff],
+  ]
+
+  def setup
+    # In case the previous test left this as true.
+    $TEST_DEBUG = false
+  end
+
+  # Print debugging information if indicated.
+  def debug(label, value, newline: false)
+    return unless $TEST_DEBUG
+    print("\n") if newline
+    printf("%15s: %s\n", label, value.inspect)
+  end
+
+  # Return the test name (retrieved from the call stack).
+  def get_test_name
+    caller.each do |x|
+      method_name = x.split(' ').last.gsub(/\W/, '')
+      return method_name if method_name.start_with?('test')
+    end
+    raise RuntimeError.new('No test method name found.')
+  end
+
+  # Perform the testing defined in the caller's block.
+  def do_test(debugging: false)
+    # Just the caller's block, unless debugging.
+    unless debugging
+      yield
+      return
+    end
+    # Wrap the caller's block with debugging information.
+    $TEST_DEBUG = true
+    test_name = get_test_name
+    debug('BEGIN', test_name, newline: true)
+    yield
+    debug('END', test_name)
+    $TEST_DEBUG = false
+  end
+
+  # Return CSV string generated from rows array and options.
+  def make_csv_s(rows: Rows, **options)
+    csv_s = CSV.generate(**options) do|csv|
+      rows.each do |row|
+        csv << row
+      end
+    end
+    csv_s
+  end
+
+  # Return filepath of file containing CSV data.
+  def csv_filepath(csv_in_s, dirpath, option_sym)
+    filename = "#{option_sym}.csv"
+    filepath = File.join(dirpath, filename)
+    File.write(filepath, csv_in_s)
+    filepath
+  end
+
+  # Return stdout and stderr from CLI execution.
+  def execute_in_cli(filepath, cli_options_s = '')
+    debug('cli_options_s', cli_options_s)
+    command = "cat #{filepath} | ruby bin/csv-filter #{cli_options_s}"
+    capture_subprocess_io do
+      system(command)
+    end
+  end
+
+  # Return results for CLI-only option (or invalid option).
+  def results_for_cli_option(option_name)
+    cli_out_s = ''
+    cli_err_s = ''
+    Dir.mktmpdir do |dirpath|
+      sym = option_name.to_sym
+      filepath = csv_filepath('', dirpath, sym)
+      cli_out_s, cli_err_s = execute_in_cli(filepath, option_name)
+    end
+    [cli_out_s, cli_err_s]
+  end
+
+  # Get and return the actual output from the API.
+  def get_via_api(csv_in_s, **api_options)
+    cli_out_s = ''
+    CSV.filter(csv_in_s, cli_out_s, **api_options) {|row| }
+    cli_out_s
+  end
+
+  # Test for invalid option.
+
+  def test_invalid_option
+    do_test(debugging: false) do
+      %w[-Z --ZZZ].each do |option_name|
+        cli_out_s, cli_err_s = results_for_cli_option(option_name)
+        assert_empty(cli_out_s)
```

Comment:
`assert_equal` is better than `assert_empty` because it shows actual text when the assertion is failed.

```suggestion
        assert_equal("", cli_out_s)
```